core/scheduler.go

```diff
@@ -331,6 +331,17 @@ func incomingMessageHandler(state *internalState, fidStore store.Singleton) even
 			default:
 				return errors.New(fmt.Sprintf("unrecognized response for controlcommand %s", incomingCommand.CommandName))
 			}
+		case "TaskMessage":
+			var taskMessage event.TaskMessageBase
+			err = json.Unmarshal(data, &taskMessage)
+			if err != nil {
+				return
+			}
+
+			t := state.taskman.GetTask(taskMessage.GetTaskId())
+			if t != nil {
+				t.SetTaskPID(taskMessage.GetTaskPID())
```

Comment:
Just a side note - once `scheduler.go` is wrapped into `task.Manager`, this call will be internal to the `task` package so `setTaskPID` should be unexported.